File: app/controllers/JsonController.java
package controllers;

import java.util.*;
import de.htwg.se.nmm.Game;
import de.htwg.se.nmm.controller.IGameController;
import de.htwg.se.nmm.model.IJunction;
import de.htwg.se.nmm.model.IPuck;
import de.htwg.se.nmm.model.IPlayer;
import de.htwg.se.nmm.aview.tui.TextUI;
import play.mvc.Controller;
import play.mvc.Result;
import play.mvc.BodyParser;
import play.libs.Json;
import play.libs.Json.*;
import views.html.tuiGame;

import services.JsonWorker;

import com.fasterxml.jackson.databind.JsonNode;
import com.fasterxml.jackson.databind.node.ObjectNode;
import com.fasterxml.jackson.databind.ObjectMapper;


/**
 * This controller contains an action to handle HTTP requests
 * to the application's home page.
 */
public class JsonController extends Controller {

    IGameController gameController = Game.getInstance().getController();

    public Result get() {
        Result jsonResult = ok(gameController.getJson()).as("application/json");
        return jsonResult;
    }

    @BodyParser.Of(BodyParser.Json.class)
    public Result changePlayerName() {
        JsonNode json = request().body().asJson();

        JsonWorker.processJson(json);

        Result jsonResult = ok(gameController.getJson()).as("application/json");
        return jsonResult;
    }

    @BodyParser.Of(BodyParser.Json.class)
    public Result process() {
        JsonNode json = request().body().asJson();

        JsonWorker.processJson(json);

        Result jsonResult = ok(gameController.getJson()).as("application/json");
        return jsonResult;
    }
}

File: app/controllers/JsonController.scala
package controllers

import javax.inject.Inject

import services.JsonWorker
import com.mohiva.play.silhouette.api._
import de.htwg.se.nmm.Game
import play.api.i18n.{ I18nSupport, MessagesApi }
import play.api.libs.json._
import play.api.mvc.{ Action, Controller }
import utils.auth.DefaultEnv
import scala.concurrent.ExecutionContext.Implicits.global

import scala.concurrent.Future

/**
 * The `Change Password` controller.
 *
 * @param messagesApi            The Play messages API.
 * @param silhouette             The Silhouette stack.
 */
class JsonController @Inject() (
  val messagesApi: MessagesApi,
  silhouette: Silhouette[DefaultEnv])
  extends Controller with I18nSupport {

  var gameController = Game.getInstance().getController

  def get = Action.async { implicit request =>
    silhouette.SecuredRequestHandler { securedRequest =>
      Future.successful(HandlerResult(Ok, Some(securedRequest.identity)))
    }.map {
      case HandlerResult(r, Some(user)) => Ok(gameController.getJson()) as "application/json"
      case HandlerResult(r, None) => Unauthorized
    }
  }

}